shed/event_streams.py

```diff
@@ -1073,81 +1073,95 @@ class BundleSingleStream(EventStream):
     def __init__(self, child, control, predicate_against=('start', 'stop'),
                  **kwargs):
         self.predicate_against = predicate_against
-        self.maxsize = kwargs.pop('maxsize', 100)
-        self.buffers = []
-        self.desc_start_map = {}
-        self.condition = Condition()
-        self.prior = ()
         self.control = control
```

Comment:
there is one issue with this. the control stream cannot be the data stream. not sure if that's a problem

Reply:
That should be fine, worst case you hand it in twice

Reply:
Just to be clear, by handing twice I assume you mean the second of these two possibilities (first won't work):

first:
```python
s1 = sin.map(somefunc)
sout = es.BundleSingleStream(s1, s1)
```
second:
```python
s1 = sin.map(somefunc)
# do same operation again
s2 = sin.map(somefunc)
sout = es.BundleSingleStream(sin, sin)
```

Reply:
Could we run a copy/deepcopy on the first one?

Reply:
maybe? If you want. Why doesn't the first one work again?